oxli/build_graph.py

```diff
@@ -0,0 +1,103 @@
+#! /usr/bin/env python2
+#
+# This file is part of khmer, http://github.com/ged-lab/khmer/, and is
+# Copyright (C) Michigan State University, 2009-2015. It is licensed under
+# the three-clause BSD license; see doc/LICENSE.txt.
+# Contact: khmer-project@idyll.org
+#
+# pylint: disable=invalid-name,missing-docstring
+"""
+Build a graph from the given sequences, save in <ptname>.
+
+% python scripts/load-graph.py <ptname> <data1> [ <data2> <...> ]
+
+Use '-h' for parameter help.
+"""
+
+import sys
+
+import khmer
+from khmer.khmer_args import (report_on_config, info, add_threading_args)
+from khmer.kfile import check_input_files, check_space
+from khmer.kfile import check_space_for_hashtable
+from oxli import functions
+
+
+def build_parser(parser):
+    add_threading_args(parser)
+    parser.add_argument('--no-build-tagset', '-n', default=False,
+                        action='store_true', dest='no_build_tagset',
+                        help='Do NOT construct tagset while loading sequences')
+    parser.add_argument('output_filename',
+                        metavar='output_presence_table_filename', help='output'
+                        ' k-mer presence table filename.')
+    parser.add_argument('input_filenames', metavar='input_sequence_filename',
+                        nargs='+', help='input FAST[AQ] sequence filename')
+    parser.add_argument('--report-total-kmers', '-t', action='store_true',
+                        help="Prints the total number of k-mers to stderr")
+    parser.add_argument('--write-fp-rate', '-w', action='store_true',
```

Comment:
No way to disable

On Tue, May 19, 2015 at 4:57 PM Jake Fenton <notifications@github.com>
wrote:

> In oxli/build_graph.py
> <https://github.com/dib-lab/khmer/pull/961#discussion_r30646001>:
>
> > +from oxli import functions
> > +
> > +
> > +def build_parser(parser):
> > +    add_threading_args(parser)
> > +    parser.add_argument('--no-build-tagset', '-n', default=False,
> > +                        action='store_true', dest='no_build_tagset',
> > +                        help='Do NOT construct tagset while loading sequences')
> > +    parser.add_argument('output_filename',
> > +                        metavar='output_presence_table_filename', help='output'
> > +                        ' k-mer presence table filename.')
> > +    parser.add_argument('input_filenames', metavar='input_sequence_
>
> filename',
> > +                        nargs='+', help='input FAST[AQ] sequence filename')
> > +    parser.add_argument('--report-total-kmers', '-t', action='store_true',
> > +                        help="Prints the total number of k-mers to stderr")
> > +    parser.add_argument('--write-fp-rate', '-w', action='store_true',
>
> turn on by default as in always on with no way to disable, or flip flop
> the arguments so the arg disables reporting?
>
> Can't imagine why you'd want to turn that off, though...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/961/files#r30646001>.
>